hw8/src/BoundedQueue.java

```diff
@@ -0,0 +1,65 @@
+import java.util.*;
+
+public class BoundedQueue <E> {
+
+    protected List<E> rep;
+    protected int size = 0;  // max size of queue
+
+    public BoundedQueue(int size) {
+        if (size > 0) {
+            this.size = size;
+            rep = new ArrayList<>();  // should we predefine ArrayList size?
+        }
+    }
+
+    public boolean isEmpty() {
+        return (rep.isEmpty());
+    }
+
+    public boolean isFull() {
+        return (rep.size() == size);
+    }
+
+    public int getCount() {
+        return rep.size();
+    }
+
+    public <E> void put(E e) {
+        if (e != null && !isFull()) {
+            rep.add(getCount()-1, e);
```

Comment:
@1000DS000000 for some reason I'm getting the follow error, which doesn't make sense to me. Thoughts?

![image](https://user-images.githubusercontent.com/11047068/47534011-d7583e00-d883-11e8-8f00-03ae252f29a3.png)